### PR TITLE
Firestore:  add Watch Support

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

**Comment by tseaver** [comment]
@crwilcox, @mcdonc  Kokoro won't run until the conflict is resolved.

**Comment by crwilcox** [comment]
@tseaver , thanks. Just merged master in. Should be good to go now.

File: firestore/google/cloud/firestore_v1beta1/bidi.py [hunk 1]
```diff
@@ -0,0 +1,559 @@
+# Copyright 2017, Google LLC
```
Comment (crwilcox) [inline firestore/google/cloud/firestore_v1beta1/bidi.py]: NOTE: this is a copy from pubsub. Plan to refactor to api_core to share.

Comment (jadekler) [reply firestore/google/cloud/firestore_v1beta1/bidi.py]: Should that license be 2018, btw?

**Comment by tseaver** [comment]
@crwilcox Maybe rebase against `master` to minimize the amount of faux diffs Kokoro sees when deciding to run tests?

**Comment by crwilcox** [comment]
rebased, fixed up watch to use transports layer, etc.

**Comment by tseaver** [comment]
W00t!, down to only one failing system test.

**Comment by mcdonc** [comment]
@googlebot I consent.

**Comment by mcdonc** [comment]
For the record, I believe a googler needs to set the "cla: yes" tag by hand on this to vacate the failure of the "cla/google" check. I'm the author of the commits it's complaining about.  I have a CLA, and if you read the error message at https://github.com/googleapis/google-cloud-python/pull/6191#issuecomment-428757051 carefully, it agrees I do.  Instead, it seems that googlebot  just a bit worried that the PR contains commits not from the direct owner of the PR repository, even though all contributors have a CLA.

I tried to set the "cla: yes" tag myself by hand but googlebot complained that it needed to be done by a googler.

**Comment by crwilcox** [comment]
@tseaver , we can't release, but we could merge this while doing conformance testing. I should have a new merge coming in that removes some `TODO` items. Unless there is a reason getting this in before figuring out conformance I have no problem keeping it open since we can't release until after anyway.

**Comment by gabisurita** [comment]
Hi! When can I expect a release from `google-cloud-firestore` including these changes?

**Comment by matej-svejda** [comment]
I would also be very much interested in knowing when this will be released. Keep up the great work guys :)

**Pull request merged**